iothub/device/src/Common/Exceptions/DeviceDisabledException.cs

```diff
@@ -1,79 +0,0 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System;
-using System.Runtime.Serialization;
-using Microsoft.Azure.Devices.Client.Extensions;
-
-namespace Microsoft.Azure.Devices.Client.Exceptions
-{
-    // TODO: #707 - This exception is not thrown by any protocol.
-
-    /// <summary>
-    /// The exception that is thrown when the device was disabled from IoT Hub.
-    /// </summary>
-    [Serializable]
-    public sealed class DeviceDisabledException : IotHubException
```

Comment:
This type was not referenced anywhere in the SDK -> no exception returned by service was mapped to this type.
@prmathur-microsoft , @sharmasejal This is a public type that we are removing.